### PR TITLE
Add netboxr tool (beta version) - third attempt

### PR DESCRIPTION
**Pull request by sajo25**
Changes made based on feedback from second PR (formatting, error detection, improving .shed.yml file, etc.)

### CODE REVIEW

File: tools/netboxr/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+description: netboxr enables automated discovery of biological process modules by network analysis
+long_description: |
+  The NetBox algorithm identifies candidate cancer-related functional modules using a data-driven, network-based approach that combines prior knowledge with a network clustering algorithm. The network modules are derived de novo, and can be used to identify new functional gene groups that cross the boundaries of curated gene sets.
+name: netboxr
+owner: sajo25
```
Comment (bgruening) [inline tools/netboxr/.shed.yml:5]: bgruening - its the TS account.

File: tools/netboxr/netboxr_xml.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            <option value="ebc">EBC</option>
+            <option value="lec">LEC</option>
+            <option value="louvain">LOUV</option>
+            <option value="leiden">LEID</option>
+        </param>
+        <param name="globalModel" type="boolean" truevalue="True" falsevalue="False" checked="True" label="Global network null model" help="Perform global network null model analysis. The global network null model calculates the empirical p-value as the number of times (over a set of iterations) the size of the largest connected component (the giant component) in the network coming from the same number of randomly selected genes equals or exceeds the size of the largest connected component in the observed network. Default: Yes" />
+        <param name="globalIterations" type="text" truevalue="True" falsevalue="False" checked="True" label="Global network null model iterations" help="Global network null model iterations. Recommended: 1000" />
+        <param name="globalNumber" type="text" truevalue="True" falsevalue="False" checked="True" label="Global network model number of genes" help="Global network null model number of genes." />
```
Comment (bgruening) [inline tools/netboxr/netboxr_xml.xml]: why is this a type="text" it can be integer isn't it?

File: tools/netboxr/netboxr_xml.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        <param name="localModel" type="boolean" truevalue="True" falsevalue="False" checked="True" label="Local network null model" help="Perform local network null model analysis. The local network null model evaluates the deviation of modularity in the observed network from modularity distribution in the random network. Default: Yes" />
+        <param name="localIterations" type="text" truevalue="True" falsevalue="False" checked="True" label="Local network null model iterations" help="Local network null model iterations. Recommended: 1000" />
+        <param name="networkPlot" type="boolean" truevalue="True" falsevalue="False" checked="True" label="Network plot" help="Output the annotated graph of netboxr output. Default: Yes" />
+        <param name="plotWidth" type="text" label="Network plot width" help="Plot width. Recommended: 9" />
```
Comment (bgruening) [inline tools/netboxr/netboxr_xml.xml]: this can also be a number and not a text, correct?

**Review by bgruening — COMMENTED** [review]
Thanks, please also see my commit.

You need to singe-quote all data and text parameters. And you can use argument - instead of name when the argument has the some form as the name :)

File: tools/netboxr/.shed.yml (hunk 1 above)
Comment (sajo25) [reply tools/netboxr/.shed.yml:5]: Do you mean that I should write "bgruening" under "owner"?

Comment (bgruening) [reply tools/netboxr/.shed.yml:5]: yes, sorry. this repo submits everything as bgruening. You can also submit your tool to https://github.com/galaxyproject/tools-iuc and then the owner is IUC.

File: tools/netboxr/netboxr_xml.xml [hunk 4]
```diff
@@ -1,35 +1,46 @@
-<tool id="netboxr" name="netboxr" version="1.6.0">
-    <description>netboxr enables automated discovery of biological process modules by network analysis.</description>
+<tool id="netboxr" name="netboxr" version="1.6.0" profile="21.05">
```
Comment (sajo25) [inline tools/netboxr/netboxr_xml.xml:1]: @bgruening The addition of "profile="21.05"" introduces some kind of bug. Do you know what the reason might be? Can the PR be merged without this?

Comment (bgruening) [reply tools/netboxr/netboxr_xml.xml:1]: I think it was the wrong spelling of geneList

Comment (sajo25) [reply tools/netboxr/netboxr_xml.xml:1]: I already changed that independently. There were also additional problems that I fixed, I'm just having issues committing the changes.

Comment (bgruening) [reply tools/netboxr/netboxr_xml.xml:1]: Mh, sorry, just wanted to help.

You need to pull the latest changes before you can commit your changes.

The profile is important as it ensures stricter isolation of tasks and more security.

**Comment by bgruening** [comment]
I think what is happening is that Galaxy tries to build the container, but that is failing.

Can you reproduce this locally with `planemo test --biocontainers .`

**Comment by sajo25** [comment]
Thanks, I'll try that! I'm just not sure why it's failing now since my last PR passed all tests.

**Comment by bgruening** [comment]
This is indeed wired. But the error is:

```
 [Aug 15 15:36:09] SERR ERROR: post-link.sh was unable to download any of the following URLs with the md5sum eef1de573f2304a4c12e64da2d9f55bd:
[Aug 15 15:36:09] SERR https://bioconductor.org/packages/3.14/data/annotation/src/contrib/GO.db_3.14.0.tar.gz
[Aug 15 15:36:09] SERR https://bioarchive.galaxyproject.org/GO.db_3.14.0.tar.gz
[Aug 15 15:36:09] SERR https://depot.galaxyproject.org/software/bioconductor-go.db/bioconductor-go.db_3.14.0_src_all.tar.gz
[Aug 15 15:36:09] SERR 
[Aug 15 15:36:09] SERR stderr:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[Aug 15 15:36:09] SERR Dload  Upload   Total   Spent    Left  Speed
[Aug 15 15:36:09] SERR 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   406  100   406    0     0   3043      0 --:--:-- --:--:-- --:--:--  3052
[Aug 15 15:36:09] SERR md5sum: WARNING: 1 computed checksum did NOT match
[Aug 15 15:36:09] SERR % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[Aug 15 15:36:09] SERR Dload  Upload   Total   Spent    Left  Speed
[Aug 15 15:36:09] SERR 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100   153    0     0    422      0 --:--:-- --:--:-- --:--:--   423
[Aug 15 15:36:09] SERR md5sum: WARNING: 1 computed checksum did NOT match
[Aug 15 15:36:09] SERR % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[Aug 15 15:36:09] SERR Dload  Upload   Total   Spent    Left  Speed
[Aug 15 15:36:09] SERR 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100   153    0     0    484      0 --:--:-- --:--:-- --:--:--   484
[Aug 15 15:36:09] SERR md5sum: WARNING: 1 computed checksum did NOT match
```

**Comment by bgruening** [comment]
And I can download the package and the md5sum looks correct:

```
(transrate) bag@bag:~/projects/code/cert-bioinf-documentation$ wget https://bioconductor.org/packages/3.14/data/annotation/src/contrib/GO.db_3.14.0.tar.gz
--2022-08-15 18:00:33--  https://bioconductor.org/packages/3.14/data/annotation/src/contrib/GO.db_3.14.0.tar.gz
Auflösen des Hostnamens bioconductor.org (bioconductor.org)… 99.86.4.110, 99.86.4.70, 99.86.4.118, ...
Verbindungsaufbau zu bioconductor.org (bioconductor.org)|99.86.4.110|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 302 Moved Temporarily
Platz: https://mghp.osn.xsede.org/bir190004-bucket01/archive.bioconductor.org/packages/3.14/data/annotation/src/contrib/GO.db_3.14.0.tar.gz [folgend]
--2022-08-15 18:00:33--  https://mghp.osn.xsede.org/bir190004-bucket01/archive.bioconductor.org/packages/3.14/data/annotation/src/contrib/GO.db_3.14.0.tar.gz
Auflösen des Hostnamens mghp.osn.xsede.org (mghp.osn.xsede.org)… 192.69.103.247, 192.69.103.248, 192.69.103.246
Verbindungsaufbau zu mghp.osn.xsede.org (mghp.osn.xsede.org)|192.69.103.247|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 30254515 (29M) [application/octet-stream]
Wird in »GO.db_3.14.0.tar.gz« gespeichert.

GO.db_3.14.0.tar.gz                                  100%[=====================================================================================================================>]  28,85M  18,9MB/s    in 1,5s    

2022-08-15 18:00:35 (18,9 MB/s) - »GO.db_3.14.0.tar.gz« gespeichert [30254515/30254515]

(transrate) bag@bag:~/projects/code/cert-bioinf-documentation$ md5sum GO.db_3.14.0.tar.gz 
eef1de573f2304a4c12e64da2d9f55bd  GO.db_3.14.0.tar.gz

```

**Comment by bgruening** [comment]
Oh its the curl issue, its not following symlinks and bioconductor has migrated the packages. Arg.

**Comment by bgruening** [comment]
Bioconda is also rebuilding packages atm, lets call it a day now and tm we might be lucky and a new GO.db package is available.

**Comment by sajo25** [comment]
Okay, sounds good. Thanks again for your help!

**Comment by sajo25** [comment]
@bgruening I made some small changes and would like to try again if that's okay with you. If it doesn't work out this time, what would be the next logical step? Also, the tests in my local Galaxy failed after I added "profile="21.05"", so that still needs to be resolved. I would appreciate any tips you may have. Thanks!

**Comment by sajo25** [comment]
@bgruening I just ran `planemo test --biocontainers .` locally and it worked. Could we try running the tests again?

**Comment by sajo25** [comment]
@bgruening I just really don't understand what is going wrong because everything works locally. In the [All tool test results](https://github.com/bgruening/galaxytools/suites/7835422839/artifacts/338462748) artifact there appears to be an error about the optparse library ("Error in library(optparse) : there is no package called \u2018optparse\u2019"), but this doesn't make sense because I am able to install and use the dependency in VirtualBox.

**Comment by bgruening** [comment]
You don't need to remove r-optparse. I just need time to look at it. Sorry, stressful here :(

**Comment by bgruening** [comment]
https://anaconda.org/bioconda/bioconductor-go.db/files go.db was also finally updated.

**Comment by sajo25** [comment]
@bgruening No worries! Maybe I can add optparse back later on. For now, though, would it be possible to merge the PR since all checks were successful?

**Comment by bgruening** [comment]
Perfect! Thanks a lot. 

If this is interesting for you, please have a look at step 3: https://galaxyproject.eu/posts/2020/08/22/three-steps-to-galaxify-your-tool/

**Pull request merged**